test/e2e/set/nodeconfig/nodeconfig_optimizations.go

```diff
@@ -456,4 +457,85 @@ var _ = g.Describe("NodeConfig Optimizations", framework.Serial, func() {
 		o.Expect(src.MatchingNodeConfigs).NotTo(o.BeEmpty())
 		o.Expect(src.BlockingNodeConfigs).To(o.BeEmpty())
 	})
+
+	g.It("should configure kernel parameters (sysctls)", func(ctx g.SpecContext) {
+		setUpRestoringCleaner := func(ctx context.Context, nc *scyllav1alpha1.NodeConfig, ignoreNotFound bool) {
+			rc := framework.NewRestoringCleaner(
+				ctx,
+				f.AdminClientConfig(),
+				f.KubeAdminClient(),
+				f.DynamicAdminClient(),
+				nodeConfigResourceInfo,
+				nc.Namespace,
+				nc.Name,
+				framework.RestoreStrategyRecreate,
+			)
+			f.AddCleaners(rc)
+			rc.DeleteObject(ctx, ignoreNotFound)
+		}
+
+		const (
+			fsAioMaxNRVar          = "fs.aio-max-nr"
+			fsAioMaxNrInitialValue = "2097152"
+			fsAioMaxNrTargetValue  = "30000000"
+		)
+
+		nc := ncTemplate.DeepCopy()
+		nc.Spec.Sysctls = []corev1.Sysctl{
+			{
+				Name:  fsAioMaxNRVar,
+				Value: fsAioMaxNrTargetValue,
+			},
+		}
+		setUpRestoringCleaner(ctx, nc, true)
+
+		// Delete any other existing NodeConfigs to ensure the test doesn't conflict with other jobs trying to configure sysctls.
+		existingNCs, err := f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for _, existingNC := range existingNCs.Items {
+			setUpRestoringCleaner(ctx, &existingNC, false)
+		}
```

Comment:
I think we could reduce this to the for loop with `ignoreNotFound` hardcoded to `true` (we do not really care that the NodeConfig we just listed has been deleted, do we?). Having a separate call for `nc` wouldn't be needed then.